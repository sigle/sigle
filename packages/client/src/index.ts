export * from './metadata/index.js';
