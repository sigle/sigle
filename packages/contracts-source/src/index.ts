import siglePublicationV0 from 'raw-loader!../../../apps/contracts/contracts/sigle-publication-v001.clar';

const contracts = { siglePublicationV0 };

export { contracts };
