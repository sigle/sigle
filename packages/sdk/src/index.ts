export * from './metadata/index.js';
export { MAX_UINT } from './lib/clarity.js';
export * from './lib/unit.js';
export * from './client/index.js';
