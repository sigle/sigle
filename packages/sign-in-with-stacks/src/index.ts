export {
  createSiwsMessage,
  type CreateSiwsMessageParameters,
  type CreateSiwsMessageReturnType,
  type CreateSiwsMessageErrorType,
} from './createSiwsMessage.js';
export { generateSiwsNonce } from './generateSiwsNonce.js';
export {
  validateSiwsMessage,
  type ValidateSiwsMessageParameters,
  type ValidateSiwsMessageReturnType,
} from './validateSiwsMessage.js';
export {
  verifySiwsMessage,
  type VerifySiwsMessageParameters,
  type VerifySiwsMessageReturnType,
} from './verifySiwsMessage.js';
