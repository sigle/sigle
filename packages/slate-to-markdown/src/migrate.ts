/**
 * Inspired by https://gist.github.com/webel/545f229fe79c2176dbaed9023de46e12
 *
 * Slate's schema has changed vastly under 2 years. The text editor is still
 * a better candidate than the other OSS editors out there, so we must live
 * with the major changes.
 *
 * Migrate a schema from the old version 0.33 to current version 0.6x
 * Inspiration taken wholly from
 * https://github.com/react-page/react-page/blob/b6c83a8650cfe9089e0c3eaf471ab58a0f7db761/packages/plugins/content/slate/src/migrations/v004.ts
 */

const migrateTextNode = (oldNode: any) => {
  const leaves = {
    text: oldNode.text,
    ...oldNode.marks?.reduce(
      (acc: any, mark: any) => ({
        ...acc,
        [mark.type === 'underlined' ? 'strikeThrough' : mark.type]: true,
      }),
      {}
    ),
  };
  return leaves;
};

const migrateLinkNode = (node: any) => {
  return {
    link: node.data.href,
    type: node.type,
    children: node.nodes?.map(migrateNode).flat() ?? [],
  };
};

const migrateImageNode = (node: any) => {
  return {
    link: node.data.src,
    caption: '',
    type: node.type,
    children: [],
  };
};

// List items needs to be wrapped in a paragraph
const migrateListItemNode = (node: any) => {
  return {
    data: node.data ?? {},
    type: 'list_item',
    children: [
      {
        type: 'paragraph',
        children: node.nodes?.map(migrateNode).flat() ?? [],
      },
    ],
  };
};

const migrateElementNode = (node: any) => {
  return {
    data: node.data ?? {},
    type: node.type,
    children: node.nodes?.map(migrateNode).flat() ?? [],
  };
};

const migrateNode = (oldNode: any) => {
  if (oldNode.object === 'text') {
    return migrateTextNode(oldNode);
  } else if (oldNode.object === 'inline' && oldNode.type === 'link') {
    return migrateLinkNode(oldNode);
  } else if (oldNode.object === 'block' && oldNode.type === 'image') {
    return migrateImageNode(oldNode);
  } else if (oldNode.object === 'block' && oldNode.type === 'list-item') {
    return migrateListItemNode(oldNode);
  } else {
    return migrateElementNode(oldNode);
  }
};

export const migrateSchema = (oldSchema: any) => {
  return oldSchema.document.nodes.map(migrateNode);
};
