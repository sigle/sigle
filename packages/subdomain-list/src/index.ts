import { fetch } from 'undici';
import { lookupProfile } from '@stacks/auth';
import * as fs from 'fs';

const appUrl = 'https://app.sigle.io';

/**
 * 1. List all the subdomains registered on Stacks
 * 2. Check if they are using Sigle in their profile
 * 3. If they allow it, add them to the list of users
 */
const start = async () => {
  let subdomainsNotFound = 0;
  let publicStoriesNotFound = 0;

  // 1. List all the subdomains registered on Stacks
  // TODO https://github.com/hirosystems/stacks-blockchain-api/issues/1097
  const subdomains = [
    'sigleapp.id.blockstack',
    'doesnotexist404.id.blockstack',
    'leopradel.id.blockstack',
  ];
  let urlsToIndex: string[] = [];

  for (const subdomain of subdomains) {
    // 2. Check if they are using Sigle in their profile
    try {
      let userProfile;
      try {
        userProfile = await lookupProfile({ username: subdomain });
      } catch (error) {
        // This will happen if there is no blockstack user with this name
        if (error.message === 'Name not found') {
          subdomainsNotFound += 1;
          continue;
        }
        throw error;
      }
      if (!userProfile.appsMeta[appUrl]) {
        continue;
      }

      const storage = userProfile.appsMeta[appUrl].storage;

      // TODO move this somewhere else?
      const res = await fetch(`${storage}publicStories.json`);
      let file: any;
      if (res.status === 200) {
        file = await res.json();
      } else if (res.status === 404) {
        publicStoriesNotFound += 1;
        continue;
      } else {
        throw new Error(
          `Unexpected status code: ${res.status} for domain ${subdomain}`
        );
      }

      // TODO allow users to disable indexing
      const storyUrls: string[] = file.stories.map(
        (story: { id: string }) => `${appUrl}/${subdomain}/${story.id}`
      );
      storyUrls.unshift(`${appUrl}/${subdomain}`);
      urlsToIndex = urlsToIndex.concat(storyUrls);
    } catch (error) {
      console.error(`Error checking ${subdomain}: ${error}`);
      process.exit(1);
    }
  }

  fs.writeFileSync(`${__dirname}/../urls.json`, JSON.stringify(urlsToIndex), {
    encoding: 'utf8',
  });

  console.log('Urls generated', { subdomainsNotFound, publicStoriesNotFound });
};

start();
