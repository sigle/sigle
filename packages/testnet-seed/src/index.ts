import { createId } from "@paralleldrive/cuid2";
import { parseBTC } from "@sigle/sdk";
import { sigleClient } from "./lib/sigle.js";
import { deployContract } from "./lib/stacks.js";

const main = async () => {
  await deployContract({
    contractName: createId(),
    codeBody: sigleClient.generatePostContract({
      collectInfo: {
        amount: 0,
        maxSupply: 100,
      },
      metadata: "ar://M5FWeRhC84c4RjX0EwsgijX7j0Jj0kftM3c33qbaHOg",
    }).contract,
    accountIndex: 1,
  });

  await deployContract({
    contractName: createId(),
    codeBody: sigleClient.generatePostContract({
      collectInfo: {
        amount: parseBTC("0.00042"),
      },
      metadata: "ar://VrLnBatCVd46e5Iy4ksspzmFUnqnFergrNGB8yx7bsU",
    }).contract,
    accountIndex: 2,
  });
};

main();
