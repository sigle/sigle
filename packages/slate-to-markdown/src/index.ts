import { serialize, defaultNodeTypes } from 'remark-slate';
import { migrateSchema } from './migrate';

export const convert = (value: any): string => {
  return migrateSchema(value)
    .map((v: any) =>
      serialize(v, {
        nodeTypes: {
          ...defaultNodeTypes,
          // TODO not working
          // delete_mark: 'underlined',
          block_quote: 'block-quote',
          ol_list: 'numbered-list',
          ul_list: 'bulleted-list',
          heading: {
            ...defaultNodeTypes.heading,
            '1': 'heading-one',
            '2': 'heading-two',
            '3': 'heading-three',
          },
        },
      })
    )
    .join('');
};
