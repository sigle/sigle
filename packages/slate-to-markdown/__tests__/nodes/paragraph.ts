import { convert } from '../../src';

it('render text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});

it('render bold text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello ',
              marks: [],
            },
            {
              object: 'text',
              text: 'bold',
              marks: [
                {
                  object: 'mark',
                  type: 'bold',
                  data: {},
                },
              ],
            },
            {
              object: 'text',
              text: ' hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});

it('render italic text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello ',
              marks: [],
            },
            {
              object: 'text',
              text: 'italic',
              marks: [
                {
                  object: 'mark',
                  type: 'italic',
                  data: {},
                },
              ],
            },
            {
              object: 'text',
              text: ' hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});

it('render underlined text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello ',
              marks: [],
            },
            {
              object: 'text',
              text: 'underlined',
              marks: [
                {
                  object: 'mark',
                  type: 'underlined',
                  data: {},
                },
              ],
            },
            {
              object: 'text',
              text: ' hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});

it('render all marks text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello ',
              marks: [],
            },
            {
              object: 'text',
              text: 'bold, italic, underlined',
              marks: [
                {
                  object: 'mark',
                  type: 'bold',
                  data: {},
                },
                {
                  object: 'mark',
                  type: 'italic',
                  data: {},
                },
                {
                  object: 'mark',
                  type: 'underlined',
                  data: {},
                },
              ],
            },
            {
              object: 'text',
              text: ' hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});

it('render code text', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'paragraph',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Hello ',
              marks: [],
            },
            {
              object: 'text',
              text: 'code',
              marks: [
                {
                  object: 'mark',
                  type: 'code',
                  data: {},
                },
              ],
            },
            {
              object: 'text',
              text: ' hello',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(convert(slateJSON)).toMatchSnapshot();
});
