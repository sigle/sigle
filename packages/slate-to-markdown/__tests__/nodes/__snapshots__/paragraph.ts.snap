// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all marks text 1`] = `
"Hello ~~***bold, italic, underlined***~~ hello
"
`;

exports[`render bold text 1`] = `
"Hello **bold** hello
"
`;

exports[`render code text 1`] = `
"Hello \`code\` hello
"
`;

exports[`render italic text 1`] = `
"Hello _italic_ hello
"
`;

exports[`render text 1`] = `
"Hello
"
`;

exports[`render underlined text 1`] = `
"Hello ~~underlined~~ hello
"
`;
